### PR TITLE
RTSP/RTP Streaming integration

### PR DESCRIPTION
**Pull request by Sudhaushu**
This source contains a feature to stream out the captured video over RTSP/RTP using H.264 video compression for remote teleoperation and telepresence applications.
The streaming feature supports the following:  
- H.264 over RTP (session negotiation via RTSP)
- RTSP digest and basic authentication 
- RTSP over secure(TLS) connection (RTSPS)  
The RTSP/RTP streaming related libraries are a part of a sub-module from the [Windows-Camera repository](https://github.com/microsoft/Windows-Camera/tree/release/NetworkVideoStreamer_1_0)
For more details on streaming [visit this page](https://github.com/microsoft/Windows-Camera/blob/release/NetworkVideoStreamer_1_0/README.md)  

Other changes include camera node code reformatting, cleanup and bug fixes related to camera sharing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,19 +21,45 @@ This is enabled by using the MFSample Publisher which published IMFSample pointe
...
+This will build the node with RTP/RTSP streaming capability. 
+To launch the node with streaming enabled, use the following command:
+
+```
+roslaunch msft_camera_rtsp test/camnodeRTSP.launch
+``` 
+After the node has been successfully launched with RTSP enabled, you can view the video on any streaming video player that supports RTSP using the url:  
+> `rtsp://<ip-address-of-machine-running-the-node>:<rtsp_port>`
+
+If using secure RTSP over TLS, use a video player that supports RTSPS with following url:
```
Comment (ooeygui) [inline README.md:51]: Is there a way to disable the insecure rtsp stream? (I didn't see that explicitly called out)

**Review by ooeygui — APPROVED** [review]
Looks great!!!!

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,6 +9,7 @@ find_package(Boost REQUIRED COMPONENTS system thread)
 set (CMAKE_CXX_STANDARD 17)
 set (CMAKE_CXX_STANDARD_REQUIRED TRUE)
 set (CMAKE_CXX_EXTENSIONS FALSE)
+set (CMAKE_CXX_FLAGS "/EHsc")
```
Comment (shschaefer) [inline CMakeLists.txt]: This should be add_compile_options(/EHsc).  It is additive to other generator settings.

File: README.md (hunk 1 above)
Comment (Sudhaushu) [reply README.md:51]: This is controlled by the "rtsps_cert_subject" parameter.
I made the parameter description more detailed to explain this.

File: CMakeLists.txt (hunk 2 above)
Comment (Sudhaushu) [reply CMakeLists.txt]: done.

**Review by ooeygui — APPROVED** [review]

**Review by ooeygui — APPROVED** [review]

**Pull request merged**